### PR TITLE
Issue 444: Added Test Data Generator Code & modified README files

### LINKED ISSUE #444: Python - Test data generator (CSV)
# Description

A python script/file which will provide a CSV of generated test data 

## Language

- [ ] Python

## Type of issue

- [ ] Feature

## Checklist:

- **For Feature**

| name            | about                                                  | labels  | assignees                                                      |
| --------------- | ------------------------------------------------------ | ------- | -------------------------------------------------------------- |
| test data generator (CSV) | Provide a simple script which can generate test data in CSV file format | feature | I would self-assign this |

**Is your feature request related to a problem? Please describe.**: I spent a while looking for test data generation options that could easily be fitted in to projects and re-used

**Describe the solution you'd like**: I would like to create a useful script which can generate test data in a CSV file, containing multiple options around the included content that the user can add/remove

### PR DESCRIPTION
**Pull request by srimontidutta**
Issue 444
Added Test Data Generator Code & modified README files following styling suggestions.
FIxes #444 
#879 
@vybhav72954 Kindly check it. Also, please update the level tag.

### CODE REVIEW

File: Python/Test Data Generator/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+House Address
+
+Latitude
+
+Longitude
+
+Country
+
+Course Comments
```
Comment (HarshCasper) [inline Python/Test Data Generator/README.md:38]: Keep it as a bulleted list. And also remove a lot of blank lines on the markdown.

File: Python/Test Data Generator/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Saving/Exporting Dataset
+
+The generated dataset consisting of the specified number of entries from the user is then exported to the local device in csv format.
+
+
+```python
```
Comment (HarshCasper) [inline Python/Test Data Generator/README.md:45]: What does this mean?

File: Python/Test Data Generator/Test data generator.ipynb [hunk 3]
```diff
@@ -0,0 +1,119 @@
+{
```
Comment (HarshCasper) [inline Python/Test Data Generator/Test data generator.ipynb:1]: Have a raw Python Script as well which would make it easy for us to review.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
We don't add level tags unless and until the PRs are merged. And please don't open new PRs if we request any change. Work on previous PR itself. 

Thanks!

File: Python/Test Data Generator/Test data generator.ipynb (hunk 3 above)
Comment (srimontidutta) [reply Python/Test Data Generator/Test data generator.ipynb:1]: So, should I include a file in .py format along with the Jupyter file?

**Comment by vybhav72954** [comment]
@srimontidutta you have not incorporated all the changes I suggested in the previous PR as well.

**Comment by vybhav72954** [comment]
Wait, there is a small mistake here!

There are 2 directories in the PR.

You have kept the older files as well. I think the branch you made for this PR was checked out from some other branch than the master.

**Comment by vybhav72954** [comment]
Please do one thing. In your branch `issue-444new` kindly delete the directory `Test Data Generator` (mind the missing underscores). The file is still in your codebase and we don't require it. Do a fresh `git add` and then a `git commit`, push the changes in the branch so that the unnecessary files are no longer in the codebase.

**Comment by srimontidutta** [comment]
@vybhav72954 I have made the suggested changes. Kindly review it.

File: Python/Test_Data_Generator/Test data generator.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
```
Comment (vybhav72954) [inline Python/Test_Data_Generator/Test data generator.py:1]: Please remove the `shebang`.

File: Python/Test_Data_Generator/Test data generator.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+# coding: utf-8
+
+# # Importing Libraries
+
+# In[1]:
```
Comment (vybhav72954) [inline Python/Test_Data_Generator/Test data generator.py:6]: I see you converted the ipynb to a py file.

This introduces multiple non required comments like `# In[1]:`

Kindly remove all such meta data comments.

File: Python/Test_Data_Generator/Test data generator.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+from random import randint
+import csv
+
+
+# # Data Generator Function
+
+# In[2]:
+
+
+def datagen(np, parameters):
```
Comment (vybhav72954) [inline Python/Test_Data_Generator/Test data generator.py:20]: Missing a docstring

File: Python/Test_Data_Generator/Test data generator.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+
+if __name__ == '__main__':
+    np = int(input(" Enter Number of Student Data:"))
+    parameters = ["Student ID", "Student Name", "Email Id", "Phone Number", "Date of Birth", "House Address",
+               "Latitude", "Longitude", "Country", "Course Comments"]
+    datagen(np, parameters)
+    print("File saved.")
+
+
```
Comment (vybhav72954) [inline Python/Test_Data_Generator/Test data generator.py:57]: Multiple blank lines that arent required. only 1 blank line is appreciated.

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Comment by vybhav72954** [comment]
@srimontidutta Any updates

**Comment by srimontidutta** [comment]
> @srimontidutta Any updates

@vybhav72954 Couldn't work on it due to exams. I'll make the PR by tomorrow.

**Comment by srimontidutta** [comment]
I have made the PR. Kindly check @vybhav72954

**Comment by vybhav72954** [comment]
> I have made the PR. Kindly check @vybhav72954

@srimontidutta Please go through all the changes requested. There are multiple things yet to be addressed

**Comment by srimontidutta** [comment]
> > I have made the PR. Kindly check @vybhav72954
> 
> @srimontidutta Please go through all the changes requested. There are multiple things yet to be addressed

@vybhav72954 I committed the same file by mistake, added the modified one right now. Kindly review.

File: Python/Test_Data_Generator/Test data generator.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+                "Email Id": fake.email(),
+                "Phone Number": phone_number,
+                "Date of Birth": random.randint(1950,2010),
+                "House Address": fake.address(),
+                "Latitude": fake.latitude(),
+                "Longitude": fake.longitude()
+            })
+"""
+Saving the geneated dataset in a csv file on your local machine.
+"""
```
Comment (vybhav72954) [inline Python/Test_Data_Generator/Test data generator.py:35]:
```py
if __name__ == '__main__':
"""
Saving the geneated dataset in a csv file on your local machine.
"""
```

File: Python/Test_Data_Generator/Test data generator.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
+## Importing Libraries
```
Comment (vybhav72954) [inline Python/Test_Data_Generator/Test data generator.py:1]: A small suggestion. Can you follow [this blog](https://dev.to/vybhav/beginners-guide-to-auto-formatting-1hpp).

Use aggressive `autopep8` (2 levels) and format this script.

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
Final Changes

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Comment by vybhav72954** [comment]
@srimontidutta Updates?

**Comment by srimontidutta** [comment]
> @srimontidutta Updates?

I'll make the PR by tomorrow @vybhav72954

**Comment by srimontidutta** [comment]
@vybhav72954 Kindly review.

**Review by vybhav72954 — APPROVED** [review]
LGTM

**Pull request merged**